services/src/modules/authentication/strategies/jwt.ts

```diff
@@ -0,0 +1,25 @@
+import fastify from 'fastify';
+import { decode } from 'jsonwebtoken';
+import logger from '../../logger';
+import { authenticationConfig } from '../../config';
+
+export default async function (request: fastify.FastifyRequest): Promise<void> {
+  const authHeader: string = request.headers.authorization;
+  if (!authHeader) return Promise.reject(new Error('Unauthorized'));
+  const token = authHeader.split(' ')[1];
+  const decodedToken = decode(token, { json: true }) as Record<string, unknown>;
```

Comment:
Third place we decode the token for the same request, by this point we must just put the decoded token on the fastify request (which is available in all places) and decode only if we can't find it on the request (so only the first one will actually decode it).
Probably just make a helper function somewhere that gets a fastify requests and returns the decoded token (it will know to check if it was already previously decoded and saved on the request, so the other places won't have to know the structure of the request context)

Reply:
Yeah. I noticed this. I'll add plugin to fastify that will set decoded jwt value on the request (if JWT exists in authorization header).